### PR TITLE
Touch UI fixes for Cocoa Press

### PR DESCRIPTION
**Pull request by marcio-cp**
Changes from Cocoa Press branch:

- Touch UI changes for Cocoa Press
    - The Cocoa Press UI is based on the Bio Touch UI, but with a new landscape mode.
    - Implemented Cocoa Press color scheme
- Made touch UI About screen vendor neutral
- Allow multi-colored custom vendor logo for Touch UI.
    - Allow custom vendor logos in the Touch UI when SHOW_CUSTOM_BOOTSCREEN is enabled.
    - Allow different layers of color in the custom logo.
- Silence compiler warnings about unused variables.
- Fixes and improvements to FTDI display routines.
    - Use floats rather than ints for HSL color conversion routines.
    - Fixed incorrect computation of bounds due to sign conversion.
    - Fixed svg2cpp.py script incorrectly outputting negative values.
- Adjustments to SERIAL_ECHOPAIR calls in FTDI library
    - Even though Marlin supports multiple pairs in SERIAL_ECHOPAIR, the compatibility macros do not.
    - These changes ensure the code compiles when the ftdi_eve_lib is used outside of Marlin

Carry-over Changes from LulzBot branch:

- Add ability to turn off the DUE MMC host bridge.
    - This functionality can be extremely slow in certain scenarios.
- Touch UI improvements to filament change and endstop states screen
- On Archim, make a backup of EEPROM data to external SPI flash chip.
    - On Due boards, Marlin emulates EEPROM using PROGMEM and that data is lost when new firmware is loaded.
    - This feature makes backups to the EEPROM on the SPI flash chip on the Archim and restores it when needed.

### CODE REVIEW

**Comment by thisiskeithb** [comment]
I have a 5" FT810CB-HY50HD arriving from China...whenever it arrives... and can't wait to see how Touch UI works.

Thank you for continuing to improve it!

**Comment by marcio-cp** [comment]
> I have a 5" FT810CB-HY50HD arriving from China...whenever it arrives... and can't wait to see how Touch UI works.
> 
> Thank you for continuing to improve it!

You're welcome, but also thank Evan from cocoapress.com for funding continued work!

File: Marlin/src/lcd/extensible_ui/lib/lulzbot/ftdi_eve_lib/extended/sound_player.cpp [hunk 1]
```diff
@@ -38,7 +38,8 @@ namespace FTDI {
 
     #if ENABLED(TOUCH_UI_DEBUG)
       SERIAL_ECHO_START();
-      SERIAL_ECHOLNPAIR("Playing note ", int(note), ", instrument ", int(effect));
+      SERIAL_ECHOPAIR("Playing note ", int(note));
+      SERIAL_ECHOPAIR(", instrument ", int(effect));
```
Comment (thinkyhead) [inline Marlin/src/lcd/extensible_ui/lib/lulzbot/ftdi_eve_lib/extended/sound_player.cpp]: Are you using compatibility layer versions of these macros?
Looks like the second one should be `SERIAL_ECHOLNPAIR`.

**Comment by thinkyhead** [comment]
Just waiting on marcio-cp/cocoa-press-marlin#1 to get merged. Otherwise this is all good.

**Pull request merged**